DRAFT.php

```diff
@@ -66,9 +66,11 @@ function clear();
 interface Sortable
 {
     /**
+     * @param (callable(mixed $a, mixed $b): void|null) $comparator The comparator used to sort this this sortable.
+     *
      * @return static A sorted copy, or $this structure sorted in-place.
      */
-    function sort(callable $comparator = null);
+    function sort(callable $comparator = null): self;
```

Comment:
`sort` is always in-place. The caller should `clone` where required.

Reply:
I also don't think `self` is correct, we should just hint `static` in the docblock.

Reply:
Self is correct in static context at this moment, type-hinting static is not possible. And I don't think it will be possible with 7.4's covariance/contravariance either. Thus having self as a native type and static as documented type is fine, in 7.4 you can safely use more specific covariant type.

Reply:
> sort is always in-place. The caller should clone where required.

By the way, in this case the original `@return` is wrong then, it says the method returns either copy or itself. If it's always in-place, it should be void maybe?

Reply:
I think `void` is correct then. :+1:

Reply:
> Thus having self as a native type and static as documented type is fine, in 7.4 you can safely use more specific covariant type.

We are targeting 7.3.1 at this stage but could bump that to 7.4 if there is good reason.

We should avoid `self`. https://3v4l.org/QZaTv

Reply:
While `self` is invariant in <7.4, you can refer to its type by name as well: https://3v4l.org/vUrCp